common/djangoapps/enrollment/tests/test_views.py

```diff
@@ -461,41 +463,69 @@ def setUp(self):
         self.client.login(username=self.USERNAME, password=self.PASSWORD)
 
-    @patch.dict(settings.FEATURES, {'EMBARGO': True})
-    def test_embargo_change_enrollment_restrict(self):
-        url = reverse('courseenrollments')
-        data = json.dumps({
+    def _generate_data(self):
+        return json.dumps({
             'course_details': {
                 'course_id': unicode(self.course.id)
             },
             'user': self.user.username
         })
 
-        # Attempt to enroll from a country embargoed for this course
-        with restrict_course(self.course.id) as redirect_url:
-            response = self.client.post(url, data, content_type='application/json')
+    def assert_access_denied(self, user_message_url):
+        """
+        Verify that the view returns HTTP status 403 and includes a URL in the response, and no enrollment is created.
+        """
+        data = self._generate_data()
+        response = self.client.post(self.url, data, content_type='application/json')
 
-            # Expect an error response
-            self.assertEqual(response.status_code, 403)
+        # Expect an error response
+        self.assertEqual(response.status_code, 403)
 
-            # Expect that the redirect URL is included in the response
-            resp_data = json.loads(response.content)
-            self.assertEqual(resp_data['user_message_url'], redirect_url)
+        # Expect that the redirect URL is included in the response
+        resp_data = json.loads(response.content)
+        self.assertEqual(resp_data['user_message_url'], user_message_url)
 
         # Verify that we were not enrolled
         self.assertEqual(self._get_enrollments(), [])
 
+
+    @patch.dict(settings.FEATURES, {'EMBARGO': True})
+    def test_embargo_change_enrollment_restrict_geoip(self):
+        """ Validates that enrollment changes are blocked if the request originates from an embargoed country. """
+
+        # Use the helper to setup the embargo and simulate a request from a blocked IP address.
+        with restrict_course(self.course.id) as redirect_url:
+            self.assert_access_denied(redirect_url)
+
+    @patch.dict(settings.FEATURES, {'EMBARGO': True})
+    def test_embargo_change_enrollment_restrict_user_profile(self):
+        """ Validates that enrollment changes are blocked if the user's profile is linked to an embargoed country. """
+
+        # Setup the embargo
+        restricted_course = RestrictedCourse.objects.create(course_key=self.course.id)
+
+        country_code = 'US'
+        country = Country.objects.create(country=country_code)
+
+        CountryAccessRule.objects.create(
+            rule_type=CountryAccessRule.BLACKLIST_RULE,
+            restricted_course=restricted_course,
+            country=country
+        )
+
+        # Update the user's profile, linking the user to the embargoed country.
+        self.user.profile.country = country_code
+        self.user.profile.save()
```

Comment:
I think you could reproduce this issue even if lines 507-518 were removed, since the error occurred when constructing the cache key for the user's profile country.  Arguably this is the more important case, since it means everyone is effectively blocked, not just embargoed students.

Would it be possible to parametrize this test so we hit both cases?  (Blocked and not blocked)